### ISSUE TITLE
Use host-specific/CPU-specific optimization for GCC meson builds

### ISSUE THREAD

**Issue by tobigithub** [grimme-lab/xtb#155]
**Is your feature request related to a problem? Please describe.**
Adding the compiler option "-O3 -march=native" may allow for faster computations on certain CPUs (here Xeon with AVX2). "-O3" alone, which is currently implemented is 50% slower. See [[GNU fortran compiler flags on stackoverflow]](https://stackoverflow.com/questions/51135567/vectorization-with-gcc-and-gfortran)

**Describe the solution you'd like**
Add the an option to compile gcc with "-O3 -march=native" Basically add line the following line to [meson.build]( https://github.com/grimme-lab/xtb/blob/master/meson/meson.build) or an additional build option (-O3 is already there).

````
 add_project_arguments('-march=native', language: 'fortran')
`````

**Describe alternatives you've considered**
NA

**If possible state how you can assist in providing data or code to to implement the feature**
See above

**Additional context**
See also results from gcc and gfortran-8  microbenchmarks [[LINK]](https://github.com/tobigithub/quantum-xtb/issues/19).

Results (1 XEON CPU@2.3 Ghz):

Intel MKL: **21.496 sec**  
gfortran-8 (netlib/openblas): "-O3 -march=native": **56.322 sec**  
gfortran-8 (netlib/openblas) "-O3": **2 min, 19.628 sec**  

Basically Intel MKL is 6-fold faster, while gfortran vectorization allows for 2-fold faster computations and the baseline with just the "-O3" compiler option and netlib/openblas is the slowest. This may or may not be anecdotal and was just tested on a single molecule and a single Xeon CPU.

**Comment by awvwgk** [comment]
Interesting point. I think we can already use the highest optimization level by configuring the build with `--optimization=3`. On the other hand, adding an option explicitly to the meson.build is not very meson-like, we do this for Intel builds already with
https://github.com/grimme-lab/xtb/blob/8226f90e7452b71f330d1279ff5d047eb0102a74/meson/meson.build#L37
but here we can be sure the instruction set is optional. I think the Intel analog to `-march=native` would be `-xHOST`, but this usually produces dedicated code for the host machine, which cannot run on other CPUs not supporting this instruction sets. We experimented with `-xSSE4.2` a while ago (note: no `-ax`) and found the resulting binary to reject all AMD processors.

To be sure, we would also have to investigate this for GCC based builds, which means, we must first produce a statically linked binary with GCC (actually we don't have to but we want to). This is no problem if you use the MKL backend (read as use the full Intel Parallel Studio and link against the `-lmkl_gf_lp64` instead of `-lgfortran`), but a bit more difficult when trying this with a pure GCC toolchain (*i.e.* my OS does not provide static gfortran libraries).

We had long discussions regarding static linking vs. dynamic linking already, see #39 and #40 for the full discussion. We generally prefer to link our binaries statically, because we have to distribute them in some way.

**Comment by awvwgk** [comment]
The easiest way to customize the build without patching the meson.build would be to use the configuration options for meson documented in their manual:
https://mesonbuild.com/Reference-tables.html#language-arguments-parameter-names

**Comment by tobigithub** [comment]
@awvwgk I agree, I like statically linked better, if it would fail you know its the program itself, especially with all those different directories, path settings etc there can be interference, or worse
on clusters, when loading many environment modules.

I am not really familiar with all the tricks during BLAS linking,  I trust the fastest EXEs are provided as binaries, but its nice to compile and link, just because. Maybe host-specific compilation is really the correct way. 

Anyway I am happy not to deal with that, its so complicated, respect :-)

**Comment by tobigithub** [comment]
@awvwgk  solution proposed, openblas xtb compilation can be improved if needed, so I am closing that, because its non-essential, can be reopened if needed. 

Add compiler flags to meson.build
````
-march=native
````
